src/contexts/ListsContext.tsx

```diff
@@ -4,29 +4,37 @@ import { Task } from "../types/Task";
 import { DefaultLists } from "../data/tasks";
 
 interface ListsContextProps {
+  currentListId: number | string | null | undefined;
+  setCurrentListId: React.Dispatch<
+    React.SetStateAction<number | string | null | undefined>
+  >;
   addList: (list: List) => void;
-  addTask: (listId: number, task: Task) => void;
-  deleteList: (id: number) => void;
-  deleteTask: (listId: number, taskId: number) => void;
+  addTask: (listId: number | string, task: Task) => void;
+  deleteList: (id: number | string) => void;
+  deleteTask: (listId: number | string, taskId: number | string) => void;
   lists: List[];
-  updateList: (id: number, updatedList: Partial<List>) => void;
+  updateList: (id: number | string, updatedList: Partial<List>) => void;
   updateTask: (
-    listId: number,
-    taskId: number,
+    listId: number | string,
+    taskId: number | string,
     updatedTask: Partial<Task>,
   ) => void;
+  getListByListId: (listId: number | string) => List | undefined;
 }
 
 const ListsContext = createContext<ListsContextProps | undefined>(undefined);
 
 export const ListsProvider = ({ children }: { children: React.ReactNode }) => {
   const [lists, setLists] = useState<List[]>(DefaultLists);
+  const [currentListId, setCurrentListId] = useState<
+    number | string | null | undefined
+  >();
 
   const addList = (list: List) => {
-    setLists((prevLists) => [...prevLists, { ...list, tasks: [] }]);
+    setLists((prevLists) => [{ ...list, tasks: [] }, ...prevLists]);
   };
 
-  const updateList = (id: number, updatedList: Partial<List>) => {
+  const updateList = (id: number | string, updatedList: Partial<List>) => {
```

Comment:
Instead of using `number | string` everywhere why not create an `interface Id = number | string` and use it everywhere.